app/controllers/users_controller.rb

```diff
@@ -24,7 +24,9 @@ def dashboard
   private
 
   def user_params
-    params.require(:user).permit(:email, :password, :password_confirmation)
+    params[:user][:email].downcase!
+    params.require(:user).permit(:email, :password)
+
```

Comment:
does this work? might be easier to do this than the before save method I created

Reply:
All tests were passing, but we should make sure we're testing all edge cases. The before save method was erroring with the validations in the test for some reason.